### PR TITLE
Refactor codebase

### PR DESCRIPTION
**Pull request by wu-jacob**
Changes:
- Migrated from js to ts
- Used vite bundler
- Refactored folder structure

### CODE REVIEW

**Comment by wu-jacob** [comment]
Feel free to checkout to the branch and run `npm run dev` to test the website. It should still look the same.

**Comment by wu-jacob** [comment]
Probably should get prettier set up some time

File: eslint.config.js
Comment (LeoShangTang) [inline eslint.config.js:1]: I can't speak to the correctness of the eslint setup, someone else can review this

File: src/assets/react.svg
Comment (LeoShangTang) [inline src/assets/react.svg:1]: Where did this random react.svg thingy come from.

I'm assuming we don't need it

File: src/assets/images/1.jpg
Comment (LeoShangTang) [inline src/assets/images/1.jpg:1]: I'm assuming this folder isn't necessary just because the events folder has the same images

File: src/assets/react.svg
Comment (wu-jacob) [reply src/assets/react.svg:1]: it was from vite's auto setup yeah we don't need it good catch

File: src/assets/images/1.jpg
Comment (wu-jacob) [reply src/assets/images/1.jpg:1]: oh yeah, idk why they had two copies lmao

Comment (LeoShangTang) [reply src/assets/images/1.jpg:1]: We should also add a git ignore for the file created for when running build

File: eslint.config.js
Comment (wu-jacob) [reply eslint.config.js:1]: we can have a discussion ab this and config it ourselves

File: src/assets/images/1.jpg
Comment (wu-jacob) [reply src/assets/images/1.jpg:1]: yuppers

**Pull request merged**